packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2341,6 +2341,35 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     assert(child.parent == this);
   }
 
+  /// Returns a matrix that maps the local coordinate system to the coordinate
```

Comment:
As documented, this doesn't really make sense. e.g. RenderSlivers and RenderSectors don't use the cartesian coordinate system for layout, so it's not clear what a Matrix4 means.

The fix, I think, is just to change the documentation to clarify that this refers to the _paint_ coordinate system, and that specifically this is just applying `applyPaintTransform` up the tree.